src/qp_klp/StandardMetagenomicWorkflow.py

```diff
@@ -49,3 +60,141 @@ def __init__(self, **kwargs):
+        sheet.Header['Application'] = 'FASTQ Only'
+        sheet.Header['Assay'] = prep_info['data_type']
+        sheet.Header['Description'] = f'prep_NuQCJob - {pid}'
+        sheet.Header['Chemistry'] = 'Default'
+        sheet.Header['SheetType'] = 'standard_metag'
+        sheet.Header['SheetVersion'] = '90'
+        sheet.Header['Investigator Name'] = 'Qiita'
+        sheet.Header['Experiment Name'] = project_name
+
+        sheet.Bioinformatics = pd.DataFrame(
+            columns=['Sample_Project', 'ForwardAdapter', 'ReverseAdapter',
+                     'library_construction_protocol',
+                     'experiment_design_description',
+                     'PolyGTrimming', 'HumanFiltering', 'QiitaID'],
+            data=[[project_name, 'NA', 'NA', 'NA', 'NA',
+                   'FALSE', 'TRUE', sid]])
+
+        df_summary = df_summary[df_summary.file_type == 'raw_forward_seqs']
+        data = []
+        regex = re.compile(r'^(.*)_S\d{1,4}_L\d{3}')
+        for k, vals in pt.iterrows():
+            k = k.split('.', 1)[-1]
+            rp = vals['run_prefix']
+            # to simplify things we will use the run_prefix as the
+            # Sample_ID as this column is a requirement for
+            # per-sample-FASTQ, it has to be unique and this can help us
+            # keep it simple for special cases (like tubeids). However,
+            # run_prefix could have appended the cell/lane info so we need
+            # to remove it, if present
+            srp = regex.search(rp)
+            if srp is not None:
+                rp = srp[1]
+            sample = {
+                'Sample_Name': k.replace('_', '.'),
+                'Sample_ID': rp.replace('.', '_'),
+                'Sample_Plate': '',
+                'I7_Index_ID': '',
+                'index': vals['index'],
+                'I5_Index_ID': '',
+                'index2': vals['index2'],
+                'Sample_Project': project_name,
+                'Well_description': '',
+                'Sample_Well': '',
+                'Lane': '1'}
+            sheet.add_sample(sample_sheet.Sample(sample))
+            _d = df_summary[
+                df_summary.filename.str.startswith(rp)]
+            if _d.shape[0] != 1:
+                ValueError(f'The run_prefix {rp} from {k} has {_d.shape[0]} '
+                           'matches with files')
+            data.append({
+                'Lane': '1', 'SampleID': rp, 'Sample_Project': project_name,
+                'Index': vals['index'], '# Reads': _d.reads.values[0]})
+
+        sheet.Contact = pd.DataFrame(
+            columns=['Email', 'Sample_Project'],
+            data=[['qiita.help@gmail.com', project_name]])
+
+        new_sample_sheet = out_path('sample-sheet.csv')
+        with open(new_sample_sheet, 'w') as f:
+            sheet.write(f, 1)
+
+        # now that we have a sample_sheet we can fake the
+        # ConvertJob folder so we are ready for the restart
```

Comment:
the "so we are ready for the restart" comment here is opaque to me--is this class only used in cases with a restart? or is there only a restart sometimes?

Reply:
Yes, it's always a restart as we already have the FASTQ files created and we don't need to start from scratch.